### ISSUE TITLE
model can't converge

### ISSUE THREAD

**Issue by YangUSTC** [nenadmarkus/pico#21]
Hi there, I'm trying to use picolrn to train my own face detectors (using AFLW dataset).
first i use genki.py to preprocess my positive samples(24384faces),then i choose around 15G background images.
When i train the model, I haven't been able to get it to converge.In the first stage,there are 4 trees.In every stage,there are max trees as picolrn has setted.
With this in mind, I was wondering whether you had any advice as to how to solve this problem?
Sorry for my poor English!
Thanks!

**Comment by nenadmarkus** [comment]
Is the default training sample (GENKI) working well? I.e., were you able to get a working cascade?
If so, the dataset you use may be too "diverse" for trees of depth 5 (default). Try increasing the tree depth (line 851 in `picolrn.c`). E.g., set `tdepth = 7`.

**Comment by YangUSTC** [comment]
Thanks for answering my question
The default training sample(GENKI) is working well(i mean about training).As you know,the GENKI dataset has a few people face,so the detection performance is not good in FDDB dataset.
I have read your paper.In the paper,your training sample is chosen in AFLW dataset，and setting the depth of trees 6.I also use AFLW dataset and set the depth of trees 6.And i haven't changed any other parameters.
Could you tell me more details about how to choose the positive and negative samples(especially in quantity) 
Buy the way,only setting tdepth=12,in the first stage,the number of tree is 1.

**Comment by nenadmarkus** [comment]
Did you check whether your data-generating script outputs good images (faces) and that the bounding box coordinates are OK? Can you send me some face images with drawn bounding boxes generated by the script?

**Comment by YangUSTC** [comment]
Thanks for answering my question
  
    I have checked this question.That's no problem(i think).I will send you some face images with drawn bounding boxes generated by the script through the annex(especially i picked some side face).I hope you can give me some advice about the training samples.






At 2017-04-11 19:25:54, "Nenad Markuš" <notifications@github.com> wrote:


Did you check whether your data-generating script outputs good images (faces) and that the bounding box coordinates are OK? Can you send me some face images with drawn bounding boxes generated by the script?

**Comment by YangUSTC** [comment]
I have already responded to your email.Please check it.

**Comment by nenadmarkus** [comment]
Yes, I can see your emails. However, there are no images attached.

**Comment by YangUSTC** [comment]
I have checked this question.That's no problem(i think).I will send you some face images with drawn bounding boxes generated by the script through the annex(especially i picked some side face).I hope you can give me some advice about the training samples.

[images.zip](https://github.com/nenadmarkus/pico/files/915882/images.zip)

**Comment by nenadmarkus** [comment]
The images seem OK. Note that pico will have problems if there's too much variation in the dataset.

Can you send me the output of `picolrn.c` when run on your data?

**Comment by YangUSTC** [comment]
[myfacefinder.zip](https://github.com/nenadmarkus/pico/files/918385/myfacefinder.zip)
OK.I used the entire AFLW dataset(24384 face)，and used genki.py to process the image(just nrands set to 3 ).Then i got 1584 trees.I have describe the problem when i first ask you this question.

**Comment by YangUSTC** [comment]
Could you please give me your training data?

**Comment by nenadmarkus** [comment]
> myfacefinder.zip

I expected the text that `picolrn.c` outputs to the console. Please send that.

> Could you please give me your training data?

The original `facefinder` cascade was produced in 2013. Thus, I do not have its training data any more.

**Comment by linsonwang** [comment]
hi, YangUSTC
     Did you finial get your training result well? 

Linson Wang

**Comment by smartadpole** [comment]
I get the same problem. I am willing to train a model, I have the right train data, but I do not know how to use the "picolrn.c". I have a puzzle about the function——"main"——in picolrn.c.

**Comment by smartadpole** [comment]
What are the parameters——argv[1], argv[2]... mean?

**Comment by smartadpole** [comment]
nobjects == 0；The program break.

**Comment by YangUSTC** [comment]
Maybe you need to read the code, argv[1] means the training data, argv[2] means the output dir. You can find these in the founction
 intlearn_with_default_parameters(char* trdata, char* dst)




At 2017-12-15 14:15:01, "长牙的小蝌蚪" <notifications@github.com> wrote:


I get the same problem. I am willing to train a model, I have the right train data, but I do not know how to use the "picolrn.c". I have a puzzle about the code——"main"——in picolrn.c.

**Comment by smartadpole** [comment]
get it.
